marconi-chain-index/src/Marconi/ChainIndex/Extract/Datum.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TupleSections  #-}
+
+module Marconi.ChainIndex.Extract.Datum where
+
+import Cardano.Api qualified as C
+import Cardano.Api.Shelley qualified as C
+import Cardano.Ledger.Alonzo.TxWits qualified as Ledger
+import Data.Map.Strict (Map)
+import Data.Map.Strict qualified as Map
+import Data.Maybe (mapMaybe)
+
+-- | Get a map of datum hash to datum from a list of transactions.
+txsPlutusDatumsMap :: Foldable f => f (C.Tx era) -> Map (C.Hash C.ScriptData) C.ScriptData
```

Comment:
I suggest to change to something like `getPlutusDatumsFromTxs` or `getPlutusResolvedDatumsFromTxs`.